src/common.rs

```diff
@@ -95,3 +95,8 @@ impl WriteConcern {
         bson
     }
 }
+
+pub fn merge_options<T: Into<bson::Document>>(document: bson::Document, options: T) -> bson::Document {
```

Comment:
I was thinking that was well when I was implementing it. I think it's worth making a PR to the bson crate for this, but in the interest of getting the driver working with 3.4, I'd suggest we don't block this merge on it and open another PR once the functionality is added to bson.